app/models/mixins/external_url_mixin.rb

```diff
@@ -0,0 +1,12 @@
+module ExternalUrlMixin
```

Comment:
Is this feature intimately tied to custom buttons?  If so, then I feel like this should be a part of the Custom Button Mixin (or at least included in that mixin instead of every class indivudually).  This also would ensure that any new classes marked as having custom buttons automatically get this functionality.

Reply:
It would expose it to all classes supporting custom buttons on the backend, but not the Automate service model side.  Would take more work on the service model side, but that could be dynamically created in the base service model if we wanted to.

Reply:
I belive custom buttons are available at more places that this. There's a list in the BZ.

I'd keep that separate for now but if we conclude that is should be the same, I will gladly do a separate PR to fix that.